packages/edit-site/src/components/dataviews/view-grid.js

```diff
@@ -1,3 +1,71 @@
-export function ViewGrid() {
-	return 'Grid';
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalGrid as Grid,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+	Card,
+	CardBody,
+	CardMedia,
+	FlexBlock,
+	Placeholder,
+} from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import FieldActions from './field-actions';
+
+export function ViewGrid( { data, fields, view, actions } ) {
+	const mediaField = fields.find(
+		( field ) => field.id === view.layoutConfig.mediaField
```

Comment:
The Grid view works by defining a "mediaField" for the current data list. Right now, the PagesList specific usage defines a default "mediaField" for the grid view (using feature image). But I'm curious to know if you think the user should be able to use which "field" to use in the "cover". (Imagine a post type that has two media kind of fields).

Also wondering whether we should automate some of this. If the user don't define anything, use the first field of type "image" or "video" or something like that.

Reply:
Why do we try to connect an image necessarily with the grid view. IMO we shouldn't care. What we should have in mind for most views(probably for all, but definitely for list and grid for now) is to support in reordering of fields.

Reply:
Personally I've seen many apps that have grid views without any media and I think it's not uncommon. I mean we could have a different view(or keep this one with a different name) that is image oriented.

Reply:
One simple thing here: As you can see in the screenshot the "actions menu" is rendered "at the bottom of the image" (this was based on a mockup by @SaxonF ) Something like that can't be done if all fields are treated equally. So this particular view (whether it should be called "grid" or "thumbnails" or something else) requires a media/cover/video

Reply:
Check the Apple notes app. It has a grid view without media. :)

Reply:
@ellatrix I wouldn't call this "without media" it's just a different kind of "media" more like "preview media" (like we'll have with BlockPatterns)

Reply:
If we end up with a prop like `mediaField`, we should update the field `enableHiding` value to false.

Reply:
> Additionally, if we end up with a prop like mediaField, we should update the field enableHiding value to false.

It's handled automatically in the PR, if a field is defined as "mediaField", it's not rendered in the remaining fields list regardless of the value of `enableHiding` or `hiddenFields`. It's done this way for two reasons:

- enableHiding is cross layout config
- switching between different layouts should keep the "hiddenFields" as is.

Reply:
It's not handled though in the `view actions` menu in `fields` sub menu. 

> switching between different layouts should keep the "hiddenFields" as is.

This is related to my other [comment](https://github.com/WordPress/gutenberg/pull/55353#discussion_r1360330204). I don't think hidden fields should be be shared between views.

Reply:
> This is related to my other https://github.com/WordPress/gutenberg/pull/55353#discussion_r1360330204. I don't think hidden fields should be be shared between views.

Mmm 🤔 How do you think a user should define its own "grid view". What things to show...?

Reply:
I think eventually when a user does actions like hiding fields, filters etc.., will have a button `save view`. So it's going to be a `grid` view with different view config values.

That all goes down to how we handle current actions of a user in a view, when they change the view type. Do we preserve the current actions(possible merge/handle differences)? Do we ask the user? Do we have a different list for the saved views that would start fresh(based on saved view config) and when in `view actions` menu merge? 🤔